src/RSCommand.cc

```diff
@@ -158,10 +172,29 @@ int AddEntity::construct_protobuf(PMGDQuery& query,
     int node_ref = get_value<int>(cmd, "_ref",
                                   query.get_available_reference());
 
+    // Modifiyng the existing properties that the user gives
+    // is a good option to make the AddNode more simple.
+    // This is not ideal since we are manupulating with user's
+    // input, but for now it is an acceptable solution.
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
```

Comment:
Is there a chance the user might add an entity without any properties? In which case you have to create a new block here?

Reply:
no need. if props is empty, nothing will be done with it.